### PR TITLE
Upgrade mongo agent

### PR DESCRIPTION
**Pull request by nadeemshahzad**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] Have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?

### CODE REVIEW

**Comment by nadeemshahzad** [comment]
@edx/devops Please review

**Comment by feanil** [comment]
@nadeemshahzad can you link to the change log for the MMS Agent so we can review it as well?  I assume that is what told you that we needed to add the group ID to the playbook?

**Comment by nadeemshahzad** [comment]
@feanil It is optional not mentioned in Change log https://docs.cloudmanager.mongodb.com/release-notes/monitoring-agent/ I am closing Secure Config PR as it is not needed and updating the https://github.com/edx/configuration/pull/4077.

**Review by feanil — APPROVED** [review]
make sure to squash your commits.

**Comment by jibsheet** [comment]
The warning at the top of cloud.mongodb.com pretty strongly implied we needed to change to the group api keys.  That's why I made the ticket instead of just running the play.  Have we reviewed the agent key docs that warning links to? (still at the top of our mongodb pages).  Is there some reason not to switch to the group token since we'd already done the work?  It seems weird to have deleted code that works with their new key strategy (which is nice because it actually supports api key rotation among other things).

**Comment by feanil** [comment]
I'm good with adding the groupID back in if it's strongly encouraged.  I just wanted to know why we were doing it.  Thanks @jibsheet.  @nadeemshahzad please review the pages @jibsheet is talking about in case there are any other changes we need to make.

**Comment by nadeemshahzad** [comment]
@feanil @jibsheet This property is necessary as monitoring and backup agents services does not work without this. It is also mentioned in configuration files.
```
# Enter your Group ID - See: cloud.mongodb.com under Settings
# This new property became a requirement in backup agent version 5.5.0.
mmsGroupId=
```
```
#
# Enter your Group ID - See: cloud.mongodb.com under Settings
# This new property became a requirement in monitoring agent version 6.0.0.
#
mmsGroupId=
```

**Review by feanil — APPROVED** [review]

**Review by jibsheet — APPROVED** [review]

**Comment by nadeemshahzad** [comment]
@feanil I need to update the role to work with 16.04. Please have a quick look on this commit. thanks

File: playbooks/roles/mongo_mms/tasks/main.yml [hunk 1]
```diff
@@ -10,30 +10,53 @@
...
 
-- name: start mms service
+- name: start mms service trusty 
   service:
     name: "{{ item.agent }}"
     state: started
   with_items: "{{ agents }}"
+  when: ansible_distribution_release == 'trusty'
+
+- name: start mms service xenial
```
Comment (feanil) [inline playbooks/roles/mongo_mms/tasks/main.yml]: Service still works with xenial so you don't need a conditional here.  You can use `service` for both xenial and trusty.

**Review by feanil — APPROVED** [review]

**Comment by nadeemshahzad** [comment]
@feanil Yes, It worked with both, removed service conditional check.

**Pull request merged**